cmd/wfx/cmd/root/cmd.go

```diff
@@ -191,53 +192,17 @@ Examples of tasks are installation of firmware or other types of commands issued
 		}
 
 		// check for socket-based activation (systemd)
-		{
-			listeners, _ := activation.Listeners()
-			n := len(listeners)
-			if n > 0 {
-				if n != 2 {
-					return fmt.Errorf("invalid fd count: %d", n)
-				}
-				log.Info().Int("count", n).Msg("Adopting sockets provided by systemd")
-				south, err := createSouthboundCollection([]string{kindHTTP.String()}, storage)
-				if err != nil {
-					return fault.Wrap(err)
-				}
-				south.servers[0].Listener = listeners[0] // use listener created by systemd
-				serverCollections = append(serverCollections, south)
-
-				north, err := createNorthboundCollection([]string{kindHTTP.String()}, storage)
-				if err != nil {
-					return fault.Wrap(err)
-				}
-				north.servers[0].Listener = listeners[1] // use listener created by systemd
-				serverCollections = append(serverCollections, north)
-			}
-		}
+		listeners, _ := activation.Listeners()
+		serverCollections = append(serverCollections, adoptListeners(listeners, storage)...)
```

Comment:
Hm, what if `adoptListeners(...)` returns `nil`? Does it make sense to just take, e.g., one of systemd's fds or it is better to discard them all?

Reply:
It works fine with nil (this is the default case if you just start `./wfx` so it's tested very well). I would opt for the all-or-nothing approach since getting only one (or more than two) socket(s) might hint at a misconfiguration.